Source/Core/VideoCommon/FPSCounter.cpp

```diff
@@ -44,22 +45,18 @@ void FPSCounter::LogRenderTimeToFile(u64 val)
 
 void FPSCounter::Update()
 {
-  const u64 time = Common::Timer::NowUs();
-  const u64 diff = time - m_last_time;
-  m_time_diff_secs = static_cast<double>(diff / 1000000.0);
+  const s64 time = Common::Timer::NowUs();
+  const s64 diff = std::max<s64>(1, time - m_last_time);
```

Comment:
Any specific reason to save 'time' as s64 now? The source is 'u64' and I'm not sure converting it to would add any precision to the 'diff' calculation.
Also, is there reason you'd make 'diff'  have a min of 1 even when a value of 0 would be more accurate?

Reply:
I used s64 because there are some times ini which `m_last_time` can be ahead of `time`, I am unsure why but this causes the dt to be HUGE. also i made the minimum 1 because otherwise you get division by 0 errors, and its obviously not possible to have a dt of 0.

Reply:
I don't see any division errors. A delta time of 0 is possible if its the most accurate number it can represent (in case the true number is closer to 0 than 1).
Now if for some reason this code run a million times within a tiny time span, it would accomunate an additional 'diff' of 1 for every time.

Reply:
It would create a division by zero error because `raw_fps` is calculated as `1.0 / m_raw_dt` and so measuring a 0 dt would completely break the running average.

Reply:
Fair enough. Then I'd make a copy of ```m_raw_dt``` which has a min value somewhat above zero to use with the FPS average calculations, and leave the base ```m_raw_dt``` 100% accurate for any external usage (e.g. the ```GetDeltaTime()``` function).

Reply:
Or just skip updating the fps average for frames where `diff` is 0. Not too sure it would make sense.

Reply:
If `m_last_time` can be ahead of time, then it's fine to make it s64 I suppose.
I think it would be good to investigate why that would happen. If I remember correctly this cose isn't even called from multiple thread cuncurrently.

Reply:
IMHO the way this is currently used it does not make sense to have a DT of zero. I would much rather have the minimum be 1µs than 0s just to clear up any potential bugs that may occur. It is very very unlikely that 0s would be more accurate than 1µs just because you would need to call it within 500ns which is just not possible within dolphin.

Reply:
The value could be 0 before and it never gave any problems AFAIK.
I added the `GetDeltaTime()` function and I currently use it in some in development PRs and I definaly consider the possibility of the dt to be 0, and actually expect it, if it's the most accurate you can get.

Reply:
Ah, if your using this in a development branch than that's a little different, I checked the repo and just didn't see any other uses of it besides logging frame times. I can change it in a bit, but it seems a little bit problematic if your relying on a microsecond resolution clock to give you times < 1µs

Reply:
I am not relying on a microsecond resolution clock to give me times < 1us. Of course nothing would likely change, but it's just technically wrong to clamp a delta time to a min of "1", whether it's 1us, or 1ms, or whatever, it's an arbitrary value, and if 0 is more accurate, it should be at zero, and the code should support it being zero.

Your code below should account for the case where hte delta time is 0, or even 1 for that matter. I'm not sure if the fps would break (shoot in the new direction) if the delta time was 1 for one frame (`diff==1` and ```m_raw_dt==0.000001```)?